File: baekjoon/silver/15651/15651.py
n, m = map(int, input().split())


def pick(n, m, arr):
    if m == 0:
        for i in range(len(arr)):
            if i == len(arr)-1:
                print(arr[i])
            else:
                print(arr[i], end=' ')
    else:
        for i in range(1, n+1):
            pick(n, m-1, arr+[i])


for i in range(1, n+1):
    pick(n, m-1, [i])

File: baekjoon/silver/15651/15651.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var (
	reader *bufio.Reader = bufio.NewReader(os.Stdin)
	writer *bufio.Writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()

	var N, M int
	fmt.Fscanf(reader, "%d %d\n", &N, &M)

	for i := 1; i <= N; i++ {
		pick(N, M-1, []int{i})
	}
}

func pick(n, m int, arr []int) {
	if m == 0 {
		for i := 0; i < len(arr); i++ {
			if i == len(arr)-1 {
				fmt.Fprintf(writer, "%d\n", arr[i])
			} else {
				fmt.Fprintf(writer, "%d ", arr[i])
			}
		}
	} else {
		for i := 1; i <= n; i++ {
			pick(n, m-1, append(arr, i))
		}
	}
}
